detection_rules/main.py

```diff
@@ -299,52 +299,48 @@ def update_lock_versions(rule_ids):
 @root.command('kibana-diff')
 @click.option('--rule-id', '-r', multiple=True, help='Optionally specify rule ID')
 @click.option('--branch', '-b', default='master', help='Specify the kibana branch to diff against')
-def kibana_diff(rule_id, branch):
+@click.option('--threads', '-t', type=click.IntRange(1), default=50, help='Number of threads to use to download rules')
+def kibana_diff(rule_id, branch, threads):
     """Diff rules against their version represented in kibana if exists."""
     from .misc import get_kibana_rules
 
     if rule_id:
-        rules = [r for r in rule_loader.load_rules(verbose=False).values() if r.id in rule_id]
+        rules = {r.id: r for r in rule_loader.load_rules(verbose=False).values() if r.id in rule_id}
     else:
-        rules = [r for r in rule_loader.load_rules(verbose=False).values() if r.metadata['maturity'] == 'production']
+        rules = {r.id: r for r in rule_loader.load_rules(verbose=False).values()
+                 if r.metadata['maturity'] == 'production'}
```

Comment:
I'm surprised there's not a function for this yet. Seems like something that's done a lot?